File: backend/entity/Student.go
//นักเรียน

package entity

import (
	"gorm.io/gorm"
	"time"
)

type Student struct {
	gorm.Model
	Student_ID 		string			`json:"student_id"`
	TitleTH  		TitleNameTH		`json:"title_th"`
	TFirst_Name 	string			`json:"t_first_name"`
	TLast_Name    	string			`json:"t_last_name"`
	TitleENG 		TitleNameENG	`json:"title_eng"`
	EFirst_Name 	string			`json:"e_first_name"`
	ELast_Name    	string			`json:"e_last_name"`
	Citizen_ID 		string			`json:"citizen_id"`
	Tel 			string			`json:"tel"`
	DateOfBirth 	time.Time		`json:"date_of_birth"`
	Gender 			Gendertype		`json:"gender"`
	Nationality 	string			`json:"nationality"`
	Email 			string			`json:"email"`
	Religious 		string			`json:"religious"`
	Student_image 	[]byte			`json:"student_image"`
	
	Attendances []Attendances `gorm:"foreignKey:StudentID" json:"attendances"`
	GuardianStudent []GuardianStudent `gorm:"foreignKey:StudentID" json:"guardian_student"`
	StudentRecords []StudentRecords `gorm:"foreignKey:StudentID" json:"student_records"`
	EducationRecords []EducationRecords `gorm:"foreignKey:StudentID" json:"education_records"`
	AssignmentSubmit []AssignmentSubmit `gorm:"foreignKey:StudentID" json:"assignment_submit"`
	Bill []Bill `gorm:"foreignKey:StudentID" json:"bill"`

	UsersID uint 	`json:"users_id"`
	AddressID uint	`json:"address_id"`
	GradeID uint	`json:"grade_id"`
	Grade *Grade `gorm:"foreignKey:GradeID" json:"grade"`
}

File: frontend/src/interfaces/Student.ts
// enums
export type TitleNameTH = "นาย" | "นางสาว" | "นาง" | "อื่นๆ";
export type TitleNameENG = "Mr." | "Ms." | "Mrs." | "Other";
export type GenderType = "Male" | "Female" | "Other";

// core
export interface StudentInterface {
  id: number,
  student_id: string,
  TitleID: number,
  title_id: null,
  t_first_name: string,
  t_last_name: string,
  e_first_name: string,
  e_last_name: string,
  citizen_id: string,
  tel: string,
  date_of_birth: string,
  gender: string,
  nationality: string,
  email: string,
  religious: string,
  student_image: null,
  attendances: null,
  guardian_student: null,
  student_records: null,
  education_records: null,
  assignment_submit: null,
  bill: null,
  users_id: 1,
  address_id: 0,
  grade_id: 1,
  grade: null
}

export interface Grade {
  id: number;
  name: string;           // เช่น ม.4/1 หรือ Year 1
  level?: string;         // optional
}

export interface LoginStudentRequest {
  username: string;
  password: string;
}
